### DISCUSSION TITLE
[2023-03-20] GUI Team Demo

### DISCUSSION THREAD

**Discussion by wdanilo** [enso-org/enso#6017] [Weekly Demo Day]
On my end:
1. Finished https://github.com/enso-org/enso/pull/5895. In particular, this PR introduces the following changes:
  a. The Performance Monitor has now "details" panel which can be used to display additional information. For example, the "draw call count" widget, when selected, displays all the names of symbols that caused draw calls in the given frame.
  b. The Performance Monitor can be paused. After pausing, a selection area appears that can be dragged left and right in order to select and inspect data points from the past.
  c. The application does not catch all JavaScript events from window. Instead, it passes them to JS visualizations, like every website should. Some JS visualizations using D3 were fixed in order not to "steal the focus".
  d. A generic way of discovering when the mouse is not handled by EnsoGL anymore was added. Whenever the mouse hovers a div not managed by EnsoGL, the cursor will automatically revert to the system one.
  e. Also, fixes https://github.com/enso-org/enso/issues/5075.

  https://user-images.githubusercontent.com/1623053/225824897-3ae9ebee-ede1-4cbe-8a33-d4d9ba13059a.mov

2. I did a [breakdown of performance related issues here](https://github.com/enso-org/enso/issues/6001)
3. I had a pretty bad week (feeling unwell and having many calls, so my performance was worse than usual :( )

**Comment by mwu-tow** [comment]
CI troubles:
* Linker error — runner got corrupted, causing troubles. 
* Issues with nightly releases — Docker update on one of the servers introduced incompatibility.
* Regressions related to linting.

Project sharing:
* Done:
  * Project ID retrieval.
  * Opening an existing project.
  * Adding project from out-of-tree location.
* To be done:
  * Importing a project bundle. 
  * Edge case handling.
  * testing on macOS.

**Comment by kazcw** [comment]
Finished #5917:
- We now spend approximately no time parsing documentation.
- We are spending less time compiling Scala.
- More backend tests are now being run against the new parser.
- The frontend has less dead code related to the old searcher.

Implemented #6019:
- Resolved a small API issue that was preventing usage of SpectorJS: #5941.
- Added a debug option that will help diagnose hardware-specific rendering performance problems: #5940.
- Simplified profiling CLI.

**Comment by farmaazon** [comment]
The week of planning: many discussions and planning for both Execution Modes and Widgets. Make tasks for Vector_Editor widget, started the first stub implementation, but not finished yet.
